### ISSUE TITLE
Image converting behaves weirdly when an image links to another image

### ISSUE THREAD

**Issue by astral-sa** [astral-sa/salr#8]
Example: http://forums.somethingawful.com/showthread.php?action=showpost&postid=443763505&forumid=44

Need to decide how we want to handle this.

On the one hand, we could just show both images. However, if someone has a thumbnail linking to a full-sized image, it gets awkward. The current approach is to happily convert the image to the linked image, assuming users won't make one image link to something completely different.

**Comment by astral-sa** [comment]
Here's another example: http://forums.somethingawful.com/showthread.php?action=showpost&postid=444814758&forumid=46

and the specific forum code causing this:

``` javascript
          setTimeout(function () {
            var a = b,
            d = j[0].value;
            if (/^https?:\/\//.test(d) && - 1 == d.indexOf('\n') && - 1 == d.indexOf('\r')) {
              var e;
              var h = /([^:]+):\/\/([^\/]+)(\/.*)?/.exec(decodeURI(d));
              if (h) {
                var f = {
                  scheme: h[1],
                  domain: h[2],
                  path: h[3] || '',
                  filename: '',
                  query: {
                  },
                  fragment: ''
                },
                h = f.path.lastIndexOf('#');
                - 1 != h && (f.fragment = f.path.substr(h + 1), f.path = f.path.substr(0, h));
                h = f.path.lastIndexOf('?');
                - 1 != h && (f.query = F(f.path.substr(h + 1)), f.path = f.path.substr(0, h));
                h = f.path.lastIndexOf('/');
                - 1 != h && (f.filename = f.path.substr(h + 1));
                e = f
              } else e = null;
              var f = h = '',
              g = !1,
              i = e.filename.lastIndexOf('.');
              - 1 != i && (h = e.filename.substr(i + 1), f = e.filename.substr(0, i));
              if (/^([^\.]+\.)?youtube(-nocookie)?\.com$/.test(e.domain) || /^([^\.]+\.)?youtu\.be$/.test(e.domain)) e.query.v ? (d = '[video type="youtube"', e.query.hd && (d += ' res="hd"'), e.fragment && (g = F(e.fragment), g.t && (d += ' start="' + parseInt(g.t, 10) + '"')), d += ']' + e.query.v + '[/video]')  : /^\/embed/.test(e.path) ? (d = '[video type="youtube"', e.query.hd && (d += ' res="hd"'), e.query.start && (d += ' start="' + parseInt(e.query.start, 10) + '"'), d += ']' + e.path.substr(e.path.lastIndexOf('/') + 1) + '[/video]')  : (d = '[video type="youtube"', e.query.hd && (d += ' res="hd"'), e.fragment && (g = F(e.fragment), g.t && (d += ' start="' + parseInt(g.t, 10) + '"')), d += ']' + e.path.substr(1) + '[/video]'),
              g = !0;
              if (!g) switch (h) {
                case 'jpg':
                case 'gif':
                case 'png':
                  if (/(www\.|i\.)imgur.com/i.test(e.domain)) switch (e =
                    '', 5 < f.length && (e = f.substr(f.length - 1)), e) {
                    case 's':
                    case 'l':
                    case 't':
                      d = '[url=' + (d.substr(0, d.lastIndexOf('/') + 1) + f.substr(0, f.length - 1) + '.' + h) + '][img]' + d + '[/img][/url]';
                      break;
                    default:
                      d = '[img]' + d + '[/img]'
                  } else d = '[img]' + d + '[/img]'
                }
            }
            c.focus();
            s(d, a, !1)
          },
          50);
```

**Comment by astral-sa** [comment]
'Fixed' in f5954c2beeaac10754779c4545224585badc02ae and refined in 8425c36ea91e72e2ccc71100e577f22ebeda4346 - remove this workaround some time after the forum JS is fixed.

**Comment by astral-sa** [comment]
I fixed this bug in the forums JS, but we'll need to keep this code in for a little while just to be on the safe side.
